src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -22,15 +22,15 @@ public static IDisposable RequestScope(this ILogger logger, HttpContext httpCont
         public static void ApplicationError(this ILogger logger, Exception exception)
         {
             logger.ApplicationError(
-                eventId: LoggerEventIds.ApplicationStartupException,
+                eventId: new EventId(LoggerEventIds.ApplicationStartupException, "ApplicationStartupException"),
```

Comment:
Same here, we should try to just replace the `LoggerEventIds` constants themselves.